### PR TITLE
[ADD] multicurrency_journal_entries module

### PR DESCRIPTION
**Pull request by matthieuchoplin**
This new module enable to post journal entries in a different currency than the currency of the company.
For instance, if your company is in GBP:

Journal entries in different currency than company

With this module, when you enter an amount_currency with a currency_id on the account_move_line, then the amount will be converted and will populate debit and credit accordingly with the rate of the day of the transaction.
I tried to cover the module with some tests.
I was stuck on the test of the onchange method, please shout if you know how to test an onchange method.

### CODE REVIEW

**Comment by fclementic2c** [comment]
Did you think to handle the auto creation of the vat move line ?

**Comment by matthieuchoplin** [comment]
I had a look at the module "Recompute tax_amount" indeed if this is what you are talking about and I believe that the module of this PR is consistent with the module that handles the tax amount.
But I will test to see if this 2 modules work well together and let you know.

Thanks for your feedback.

**Comment by matthieuchoplin** [comment]
@fclementic2c I just tested with the module "Recompute tax_amount" and the module "Multicurrency Journal Entries", the tax_amount is correctly updated according to the debit and credit converted from the value of the fields amount_currency and currency_id.
I am not sure if this is what you were talking about. If not, please let me know your use case and I will add it to the unit tests.

**Comment by fclementic2c** [comment]
No, I was thinking about when you poste a manual entry and select a tax id
then the vat entry line is created automatically...
Do you see what I mean or do you need a example ?

_camptocamp_

INNOVATIVE SOLUTIONS
BY OPEN SOURCE EXPERTS

_Frédéric Clementi_
Project Manager
Business Solutions

+41 21 619 10 41

www.camptocamp.com

2015-04-16 16:40 GMT+02:00 Matthieu Choplin notifications@github.com:

**Comment by matthieuchoplin** [comment]
I do not know what you are talking about.. Sorry.
Can you tell me which modules need to installed for the use case and what use case it is please?

**Comment by matthieuchoplin** [comment]
I do not know why the runbot indicates that the build has failed: https://runbot.odoo-community.org/runbot/static/build/2926560-192-5a026a/logs/job_20_test_all.txt
=> there is not error in the logs.

File: multicurrency_journal_entries/README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+This module was written to extend the functionality of the e-invoicing module
+(account) in order to be able to create journal entries in a currncy different
+than the currency of the company
+
+Installation
+============
+
+To install this module, you need to:
+
+* clone the branch 8.0 of the repository https://github.com/OCA/account-closing
```
Comment (fclementic2c) [inline multicurrency_journal_entries/README.rst]: - clone the branch 8.0 of the repository https://github.com/OCA/account-financial-tools

File: multicurrency_journal_entries/README.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Multicurrency Journal Entries
+=============================
+
+This module was written to extend the functionality of the e-invoicing module
+(account) in order to be able to create journal entries in a currncy different
```
Comment (fclementic2c) [inline multicurrency_journal_entries/README.rst]: +(account) in order to be able to create journal entries in a currency different

**Comment by fclementic2c** [comment]
I did a tests and it works fine if you post entry from the Journal entries menu. 
However I noticed it is not implemented if you post entries from Journal items menu. (This is why we did not understand each other in previous posts). It is a consistency issue but it will not be as easy to implement the feature here as well. 
What do you think ?

**Comment by matthieuchoplin** [comment]
Thank you for your fixes in the README.rst file.

When I create an entry through the Journal items menu, it is not possible to enter an amount in an other currency:
Image of journal items line
I did not change that behaviour.

Do you think that it is relevant to add 2 new fields (amount_currency and currency_id) on this tree view? The feature is going to work when both the currency_id and amount_currency fields have been populated. So, adding these fields will do the job but I am not keen to add these here because I am not sure of the implications.

**Comment by matthieuchoplin** [comment]
Feedback: it has been more than a month that I have been using this module in production.
Apart from a rounding issue that was spotted and fixed at the beginning, no issue has been reported so far.

**Comment by matthieuchoplin** [comment]
@mdietrichc2c : I would be glad if you can review this module. Let us discuss during the OdooExperience days if you want.

**Pull request closed without merging**